yt/frontends/stream/data_structures.py

```diff
@@ -363,6 +366,14 @@ def _is_valid(cls, *args, **kwargs):
     def _skip_cache(self):
         return True
 
+    def _find_particle_types(self):
+        particle_types = []
```

Comment:
we can just use `set([])` here directly, and just use `.add` instead of `append`.  You'll still need to convert back to a tuple below.